### PR TITLE
Updating package definition, minimum platform versions as well as dependencies

### PR DESCRIPTION
**Pull request by Muhieddine-El-Kaissi**
1. Updating package definition to 5.10
2. Updating minimum platform versions to iOS 15 & OS 14
3. Updating swift version to 5.10
4. Cleaning up RXswift lint
5. Updating swiftlint & swift format as well as applying fixes to the code

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,13 +9,16 @@ jobs:
...
+        destination: ["platform=iOS Simulator,OS=17.4,name=iPhone 15"]
         scheme: ["Thumbprint"]
-        xcode: ["/Applications/Xcode_15.2.app"]
+        xcode: ["/Applications/Xcode_15.3.app"]
     steps:
       - name: 📥 Checkout
         uses: actions/checkout@v2
 
+      - name: Select Xcode
+        run: sudo xcode-select -s "/Applications/Xcode_15.3.app"
```
Comment (scottasoutherland) [inline .github/workflows/ci.yml:20]: I don't think you can do this ? These are githubs machines, not ours.

Comment (Muhieddine-El-Kaissi) [reply .github/workflows/ci.yml:20]: it's the work around proposed in https://stackoverflow.com/a/73896628/695599 and it fixed the lint issue we are seeing

Comment (scottasoutherland) [reply .github/workflows/ci.yml:20]: it's very strange to me that you can run sudo at all here 🤔

**Review by dwroth — COMMENTED** [review]
Code changes look good to me.  I don't have much context on the CI stuff, so I'll let Scott approve.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Muhieddine-El-Kaissi) [reply .github/workflows/ci.yml]: yeah we need sudo and it's proposed by the actions :)
Screenshot 2024-04-25 at 12 12 07 PM

Comment (scottasoutherland) [reply .github/workflows/ci.yml:20]: it's not proposed by the actions, but by xcode-select. It doesn't know it's run from github.

Comment (scottasoutherland) [reply .github/workflows/ci.yml:20]: I guess they say it's fine 🤷‍♂️ 

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#administrative-privileges

Comment (Muhieddine-El-Kaissi) [reply .github/workflows/ci.yml]: if we don't do that we get lint failure. maybe we can force merge and on the next pr we won't have this issue as we did already upgraded to xcode 15.3? 
Screenshot 2024-04-25 at 12 17 26 PM

**Review by scottasoutherland — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (scottasoutherland) [reply .github/workflows/ci.yml:20]: The vms aren't dedicated though, so i think you'd run into it again. I think eventually you'd be able to drop it when 5.10 is the default they have etc.

**Pull request merged**